### PR TITLE
update pessimistic transaction

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Update pessimistic transaction.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3177
- Other reference link(s):

### CODE REVIEW

**Comment by ran-huang** [comment]
@jackysp @yikeke PTAL. Thanks~

File: pessimistic-transaction.md [hunk 1]
```diff
@@ -7,37 +7,45 @@ aliases: ['/docs/dev/reference/transactions/transaction-pessimistic/']
 
 # TiDB Pessimistic Transaction Model
 
-In versions before 3.0.8, TiDB implements the optimistic transaction mode by default, in which the transaction commit might fail because of transaction conflict. To make sure that the commit succeeds, you need to modify the application and add an automatic retry mechanism. You can avoid this issue by using the pessimistic transaction mode of TiDB.
+To make the usage of TiDB closer to traditional databases and reduce the cost of migration, starting from v3.0, TiDB supports the pessimistic transaction model on top of the optimistic transaction model. This document describes the features of the TiDB pessimistic transaction model.
 
-## Usage
+> **Note:**
+>
+> Starting from v3.0.8, the newly created TiDB clusters use the pessimistic transaction model by default. However, this does not affect your clusters if you upgrading from v3.0.7 or earlier to v3.0.8 (and later). In other words, **only newly created clusters default to using the pessimistic transaction model**.
```
Comment (yikeke) [inline pessimistic-transaction.md:14]:
```suggestion
> Starting from v3.0.8, newly created TiDB clusters use the pessimistic transaction model by default. However, this does not affect your existing cluster if you upgrade it from v3.0.7 or earlier to v3.0.8 or later. In other words, **only newly created clusters default to using the pessimistic transaction model**.
```

File: pessimistic-transaction.md [hunk 2]
```diff
@@ -7,37 +7,45 @@ aliases: ['/docs/dev/reference/transactions/transaction-pessimistic/']
...
 
-To apply the pessimistic transaction mode, choose any of the following three methods that suits your needs:
+## Switch transaction mode
 
-- Execute the `BEGIN PESSIMISTIC;` statement to allow the transaction to apply the pessimistic transaction mode. You can write it in comment style as `BEGIN /*!90000 PESSIMISTIC */;` to make it compatible with the MySQL syntax.
+You can set the transaction mode by configuring the [`tidb_txn_mode`](/tidb-specific- system-variables.md#tidb_txn_mode) system variable. The following command sets all explicit transactions (that is, non-autocommit transactions) executed by newly created sessions in the cluster to the pessimistic transaction model:
```
Comment (yikeke) [inline pessimistic-transaction.md:18]:
```suggestion
You can set the transaction mode by configuring the [`tidb_txn_mode`](/tidb-specific-system-variables.md#tidb_txn_mode) system variable. The following command sets all explicit transactions (that is, non-autocommit transactions) executed by newly created sessions in the cluster to the pessimistic transaction mode:
```

File: pessimistic-transaction.md [hunk 3]
```diff
@@ -81,6 +89,33 @@ Pessimistic transactions in TiDB behave similarly to those in MySQL. See the min
...
+
+    > **Note:**
+    >
+    > In this isolation level, DML operations are performed based on the latest committed data. The behavior is the same as MySQL, but differs from the optimistic transaction mode in TiDB. See [Difference between TiDB and MySQL Repeatable Read](/transaction-isolation-levels.md#difference-between-tidb-and-mysql-repeatable-read).
+
+- [Read Committed](/transaction-isolation-levels.md#read-committed-isolation-level). You can set this isolation level using the [`SET TRANSACTION`](/sql-statements/sql-statement-set-transaction.md) statement.
+
+## Pipelined locking process
+
+Adding a pessimistic lock requires writing data into TiKV. The transaction can only be returned after commit and apply through Raft. Therefore, compared with optimistic transactions, the pessimistic transaction mode inevitably has higher latency.
```
Comment (yikeke) [inline pessimistic-transaction.md:106]:
```suggestion
Adding a pessimistic lock requires writing data into TiKV. The transaction can only be returned after commit and apply through Raft. Therefore, compared with optimistic transactions, the pessimistic transaction mode inevitably has higher latency.
```
"The transaction can only be returned" sounds strange. Please confirm with the author **what is returned** after commit and apply through Raft. @ran-huang  I can't tell from the original Chinese text, which is:

`加悲观锁需要向 TiKV 写入数据，要经过 Raft 提交并 apply 后才能返回，相比于乐观事务，不可避免的会增加部分延迟。`

PTAL @youjiali1995

**Comment by yikeke** [comment]
Please resolve comments above, rest LGTM

File: pessimistic-transaction.md (hunk 3 above)
Comment (ran-huang) [reply pessimistic-transaction.md:106]: Confirmed with the author: `给 TiDB 返回加锁成功的响应`
@yikeke PTAL at my edit. Thanks~

Comment (yikeke) [reply pessimistic-transaction.md:106]: lgtm

**Pull request merged**